packages/next/export/worker.js

```diff
@@ -130,10 +126,10 @@ process.on(
           }
         }
 
-        if (curRenderOpts.amphtml && query.amp) {
+        if (curRenderOpts.inAmpMode) {
           await validateAmp(html, path)
         }
-        if (curRenderOpts.hasAmp) {
+        if (curRenderOpts.hasAmpVersion) {
```

Comment:
This logic isn't correct, right? `inAmpMode` implies the page is already AMP, so checking `hasAmpVersion` is irrelevant. This seems like legacy code (dirty amp days)... 🤔

Am I misunderstanding something?

Reply:
The `inAmpMode` is separate from this. This is checking if it is a hybrid page and needs to render the AMP version thus `hasAmpVersion`

Reply:
Ah -- I'm still not fond of these names then:
```js
const hasAmpVersion = !inAmpMode && ampState.enabled
```

Has amp version means amp is enabled for the page, but is false when already being treated as amp? I don't like this. 🤔

Reply:
Yeah, I agree the naming is weird. Will tweak it some more